hirundo/_http.py

```diff
@@ -7,7 +7,10 @@
 
 def raise_for_status_with_reason(response: Response):
     try:
-        response.reason = response.json().get("reason", None)
+        if response.status_code >= 400:
+            response.reason = response.json().get("reason", None)
+            if response.reason is None:
+                response.reason = response.json().get("detail", None)
```

Comment:
This handles more situations where the server returns errors. `detail` errors are provided by `pydantic`